### PR TITLE
Arrange all examples

### PR DESCRIPTION
**Pull request by nirarg**
Fixes: [OCM-6265](https://issues.redhat.com//browse/OCM-6265)
Fixes: [OCM-5925](https://issues.redhat.com//browse/OCM-5925)
Fixes: [OCM-5921](https://issues.redhat.com//browse/OCM-5921)
Fixes: [OCM-6266](https://issues.redhat.com//browse/OCM-6266)
Fixes: [OCM-4544](https://issues.redhat.com//browse/OCM-4544)
Fixes: [OCM-6267](https://issues.redhat.com//browse/OCM-6267)
Fixes: [OCM-6269](https://issues.redhat.com//browse/OCM-6269)

### CODE REVIEW

File: examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/main.tf [hunk 1]
```diff
@@ -1,18 +1,22 @@
+locals {
+  account_role_prefix  = "${var.cluster_name}-account"
```
Comment (gdbranco) [inline examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/main.tf:2]: Any reason for these kind of prefixes?

File: examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/variables.tf [hunk 2]
```diff
@@ -1,23 +1,14 @@
...
-  description = "Create the vpc subnets in only one AZ"
-  type        = bool
-  default     = true
-}
-
-variable "machine_cidr" {
-  type    = string
-  default = "10.0.0.0/16"
+  type        = string
+  description = "Name of the cluster. Cannot exceed 15 characters in length. After the creation of the resource, it is not possible to update the attribute value."
```
Comment (gdbranco) [inline examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/variables.tf:13]: This length is changing soon, I think we shouldn't explicitly mention the max and let the API

File: examples/rosa-classic-public-with-byo-vpc/variables.tf [hunk 3]
```diff
@@ -1,23 +1,14 @@
...
-  description = "Create the vpc subnets in only one AZ"
-  type        = bool
-  default     = true
-}
-
-variable "machine_cidr" {
-  type    = string
-  default = "10.0.0.0/16"
+  type        = string
+  description = "Name of the cluster. Cannot exceed 15 characters in length. After the creation of the resource, it is not possible to update the attribute value."
```
Comment (gdbranco) [inline examples/rosa-classic-public-with-byo-vpc/variables.tf:13]: Same comment from prior name length

File: examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/variables.tf (hunk 2 above)
Comment (nirarg) [reply examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/variables.tf:13]: Ok, I will remove this from all relevant places

File: examples/rosa-classic-public-with-shared-vpc/main.tf [hunk 4]
```diff
@@ -104,25 +108,34 @@ module "rosa_cluster_classic" {
...
-  replicas                     = var.replicas
-  installer_role_arn           = module.account_iam_resources.account_roles_arn["Installer"]
-  support_role_arn             = module.account_iam_resources.account_roles_arn["Support"]
-  controlplane_role_arn        = module.account_iam_resources.account_roles_arn["ControlPlane"]
-  worker_role_arn              = module.account_iam_resources.account_roles_arn["Worker"]
   oidc_config_id               = module.oidc_config_and_provider.oidc_config_id
   aws_subnet_ids               = module.shared-vpc-policy-and-hosted-zone.shared_subnets
-  multi_az                     = true
+  multi_az                     = length(module.vpc.availability_zones) > 1
+  replicas                     = length(module.vpc.availability_zones)
```
Comment (gdbranco) [inline examples/rosa-classic-public-with-shared-vpc/main.tf:116]: Did that work? I'm under the impression that it needs 2 per availability zone :thinking:

File: examples/rosa-classic-public-with-shared-vpc/variables.tf [hunk 5]
```diff
@@ -1,57 +1,38 @@
...
 
-variable "account_role_prefix" {
-  type    = string
-  default = null
-}
-
-variable "operator_role_prefix" {
-  type    = string
-  default = null
+variable "shared_vpc_aws_access_key_id" {
```
Comment (gdbranco) [inline examples/rosa-classic-public-with-shared-vpc/variables.tf:16]: Not too much familiar with shared vpc, why is this needed?

File: examples/rosa-classic-public-with-unmanaged-oidc/variables.tf [hunk 6]
```diff
@@ -1,12 +1,14 @@
...
   validation {
     condition     = can(regex("^[0-9]*[0-9]+.[0-9]*[0-9]+.[0-9]*[0-9]+$", var.openshift_version))
     error_message = "openshift_version must be with structure <major>.<minor>.<patch> (for example 4.13.6)."
   }
 }
 
 variable "cluster_name" {
-  type = string
+  type        = string
+  description = "Name of the cluster. Cannot exceed 15 characters in length. After the creation of the resource, it is not possible to update the attribute value."
```
Comment (gdbranco) [inline examples/rosa-classic-public-with-unmanaged-oidc/variables.tf:13]: Same comment from other name length

File: examples/rosa-classic-public-with-shared-vpc/main.tf (hunk 4 above)
Comment (nirarg) [reply examples/rosa-classic-public-with-shared-vpc/main.tf:116]: replicas is the number of worker machines, I'm not familiar with such limitation.
Maybe you confuse with subnets count, which require two subnets in availability zone (one public and one private)?

File: examples/rosa-classic-public-with-shared-vpc/variables.tf (hunk 5 above)
Comment (nirarg) [reply examples/rosa-classic-public-with-shared-vpc/variables.tf:16]: The shared VPC is created on different AWS account.
This is the way to path the credentials of this account, in order to create all relevant resources in it.
The AWS account of the cluster is the passed as the default AWS account

File: modules/shared-vpc-policy-and-hosted-zone/main.tf [hunk 7]
```diff
@@ -38,22 +38,16 @@ resource "aws_iam_policy" "shared_vpc_policy" {
       {
         Effect = "Allow"
         Action = [
-          "route53:GetHostedZone",
           "route53:ChangeResourceRecordSets",
-          "route53:ChangeTagsForResource",
+          "route53:ListHostedZones",
```
[the remark this answers is not in the document]
Comment (gdbranco) [reply modules/shared-vpc-policy-and-hosted-zone/main.tf:42]: `ocm get /api/clusters_mgmt/v1/aws_inquiries/sts_policies | grep shared
      "id": "shared_vpc_openshift_ingress_operator_cloud_credentials_policy",`
      
      I see this exists

File: examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/variables.tf (hunk 2 above)
Comment (nirarg) [reply examples/rosa-classic-public-with-byo-vpc-byo-iam-byo-oidc/variables.tf]: removed

File: examples/rosa-classic-public-with-byo-vpc/variables.tf (hunk 3 above)
Comment (nirarg) [reply examples/rosa-classic-public-with-byo-vpc/variables.tf]: removed

File: examples/rosa-classic-public-with-idp-machine-pools/variables.tf [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
   validation {
     condition     = can(regex("^[0-9]*[0-9]+.[0-9]*[0-9]+.[0-9]*[0-9]+$", var.openshift_version))
     error_message = "openshift_version must be with structure <major>.<minor>.<patch> (for example 4.13.6)."
   }
 }
 
 variable "cluster_name" {
-  type = string
+  type        = string
+  description = "Name of the cluster. Cannot exceed 15 characters in length. After the creation of the resource, it is not possible to update the attribute value."
```
[the remark this answers is not in the document]
Comment (nirarg) [reply examples/rosa-classic-public-with-idp-machine-pools/variables.tf]: removed

File: examples/rosa-classic-public-with-unmanaged-oidc/variables.tf (hunk 6 above)
Comment (nirarg) [reply examples/rosa-classic-public-with-unmanaged-oidc/variables.tf]: removed

File: examples/rosa-classic-public/variables.tf [hunk 9]
```diff
@@ -1,12 +1,14 @@
...
   validation {
     condition     = can(regex("^[0-9]*[0-9]+.[0-9]*[0-9]+.[0-9]*[0-9]+$", var.openshift_version))
     error_message = "openshift_version must be with structure <major>.<minor>.<patch> (for example 4.13.6)."
   }
 }
 
 variable "cluster_name" {
-  type = string
+  type        = string
+  description = "Name of the cluster. Cannot exceed 15 characters in length. After the creation of the resource, it is not possible to update the attribute value."
```
[the remark this answers is not in the document]
Comment (nirarg) [reply examples/rosa-classic-public/variables.tf]: removed

File: modules/shared-vpc-policy-and-hosted-zone/main.tf (hunk 7 above)
[the remark this answers is not in the document]
Comment (oriAdler) [reply modules/shared-vpc-policy-and-hosted-zone/main.tf:42]: We use the policy defined here https://github.com/openshift/managed-cluster-config/blob/master/resources/sts/4.15/shared_vpc_openshift_ingress_operator_cloud_credentials_policy.json for the ingress operator policy.
The other policy is provided in the docs, and the user has to create it in the shared VPC account, outside of the ROSA flow.

File: examples/rosa-classic-public-with-shared-vpc/variables.tf (hunk 5 above)
Comment (gdbranco) [reply examples/rosa-classic-public-with-shared-vpc/variables.tf:16]: Oh this is to create the shared vpc
